metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart

```diff
@@ -349,12 +341,122 @@ void main() {
         final projectMetrics = notifier.projectsMetricsTileViewModels.first;
         final buildResultMetrics = projectMetrics.buildResultMetrics;
 
+        final viewModels = buildResultMetrics.buildResults;
+
+        expect(viewModels, hasLength(lessThanOrEqualTo(maxNumberOfBuilds)));
+      },
+    );
+
+    test(
+      "loads a build result metric with the maximum build duration from the latest number of builds to load for chart metrics build results",
+      () async {
+        const expectedMaximumDuration = Duration(days: 3);
+
+        final buildResults = [
+          createBuildResult(
+            BuildStatus.successful,
+            duration: const Duration(days: 2),
+          ),
+          createBuildResult(
+            BuildStatus.successful,
+            duration: expectedMaximumDuration,
+          ),
+        ];
+
+        final dashboardMetrics = DashboardProjectMetrics(
+          projectId: 'id',
+          buildResultMetrics: BuildResultMetric(
+            buildResults: buildResults,
+          ),
+        );
+        when(receiveProjectMetricsMock.call(any)).thenAnswer(
+          (_) => Stream.value(dashboardMetrics),
+        );
+
+        final notifier = ProjectMetricsNotifier(receiveProjectMetricsMock);
+        await setUpNotifier(notifier: notifier, projects: projects);
+
+        final projectMetrics = notifier.projectsMetricsTileViewModels.first;
+        final buildResultMetrics = projectMetrics.buildResultMetrics;
+
         final maximumDuration = buildResultMetrics.maxBuildDuration;
 
         expect(maximumDuration, equals(expectedMaximumDuration));
       },
     );
 
+    test(
+      "loads a build result metric with null builds date ranges if the build results are empty",
+      () async {
+        const dashboardMetrics = DashboardProjectMetrics(
+          projectId: 'id',
+          buildResultMetrics: BuildResultMetric(),
+        );
+        when(receiveProjectMetricsMock.call(any)).thenAnswer(
+          (_) => Stream.value(dashboardMetrics),
+        );
+
+        final notifier = ProjectMetricsNotifier(receiveProjectMetricsMock);
+        await setUpNotifier(notifier: notifier, projects: projects);
+
+        final projectMetrics = notifier.projectsMetricsTileViewModels.first;
+        final buildResultMetrics = projectMetrics.buildResultMetrics;
+
+        final firstBuildDate = buildResultMetrics.metricPeriodStart;
+        final lastBuildDate = buildResultMetrics.metricPeriodEnd;
```

Comment:
```suggestion
        final metricPeriodStart = buildResultMetrics.metricPeriodStart;
        final metricPeriodEnd = buildResultMetrics.metricPeriodEnd;
```
What do you think? Also, should we update the test description to match new namings? The same suggestion for the test below.